File: December 2022/18 December 2022/Java/A Game of LCM.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            int N = Integer.parseInt(read.readLine());
            Solution ob = new Solution();
            System.out.println(ob.maxGcd(N));
        }
    }
}
// } Driver Code Ends

class Solution {
    long gcd(long A, long B) {
        if (B == 0) return A;
        return gcd(B, A % B);
    }
    long lcm(int N) {
        if (N == 4) return 12;
        long a = N * (N - 1);
        long b = a;
        b *= (N - 2) / gcd(b, N - 2);
        b *= (N - 3) / gcd(b, N - 3);
        long c = a, f = 0;
        for (int i = N - 2; i >= 1; i--) {
            if (gcd(c, i) == 1) {
                a *= i;
                c *= i;
                f++;
            }
            if (f == 2) break;
        }
        return Math.max(a, b);
    }
    long maxGcd(int N) {
        if(N==4)return 12;
        return Math.max(lcm(N), lcm(N-1));
    }
}

File: Problems & Solutions/A/A Game of LCM/18 December 2022/Python/A Game of LCM.py
class Solution:    
    def gcd(self,A,B):
        if(B==0):
            return A
        return self.gcd(B,A%B) 
    def lcm(self,N):
        if N==4:
            return 12
        a=N*(N-1)
        b=a
        b*=(N-2)//self.gcd(b,N-2)
        b*=(N-3)//self.gcd(b,N-3)
        c=a
        f=0
        for i in range(N-2,0,-1):
            if self.gcd(c,i)==1:
                a*=i
                c*=i
                f+=1
            if f==2:
                break
        return max(a,b)
    def maxGcd(self,N):
        if N==4:
            return 12
        return max(self.lcm(N),self.lcm(N-1)) 

#{ 
 # Driver Code Starts
#Initial Template for Python 3

import math
        
if __name__=='__main__':
    t=int(input())
    for _ in range(t):
        N=(int)(input())
        ob=Solution()
        print(ob.maxGcd(N))
# } Driver Code Ends
